flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/SinkModifyOperation.java

```diff
@@ -98,7 +100,7 @@ public SinkModifyOperation(
             boolean overwrite,
             Map<String, String> dynamicOptions,
             ModifyType modifyType) {
-        this.contextResolvedTable = contextResolvedTable;
+        this.contextResolvedTable = withDynamicOptions(contextResolvedTable, dynamicOptions);
```

Comment:
It seems we are mixing the `api` module with the `planner` module here. Ideally, all the logic should go into the planner. Can we perform the merging of options there?

Reply:
I can move it later but then there will be multiple code paths that perform merging.